apps/deploy-web/src/components/new-deployment/ManifestEdit.tsx

```diff
@@ -136,7 +173,7 @@ export const ManifestEdit: React.FunctionComponent<Props> = ({ editedManifest, s
   async function handleCreateClick(deposit: number, depositorAddress: string) {
     setIsCreatingDeployment(true);
 
-    const sdl = selectedSdlEditMode === "yaml" ? editedManifest : (sdlBuilderRef.current?.getSdl() as string);
+    const sdl = selectedSdlEditMode === "yaml" ? editedManifest as string : (sdlBuilderRef.current?.getSdl() as string);
```

Comment:
This kind of type cast is dangerous and there is a way to make it safe.

```typescript
const sdl = (selectedSdlEditMode === "yaml" && editedManifest) || sdlBuilderRef.current?.getSdl();

if (!sdl) {
   setIsCreatingDeployment(false);
   return;
}
```

Reply:
ok